doc/nrf/protocols/matter/end_product/watchdog.rst

```diff
@@ -0,0 +1,188 @@
+
+  1. Create a Matter Watchdog source without any arguments.
+
+     For example:
+
+     .. code-block:: c++
+
+        Nrf::Watchdog::WatchdogSource myWatchdog;
+
+  2. Enable the Global Watchdog module:
+
+    .. code-block:: c++
+
+        Nrf::Watchdog::Enable();
+
+  3. Call the ``Feed()`` method at the specific place in the code where you want to prevent code blocking:
+
+     .. code-block:: c++
+
+        myWatchdog.Feed();
+
+
+Enabling and disabling the Watchdog peripheral
+**********************************************
+
+The Global Watchdog used in Matter Samples is a single peripheral that operates independently of the CPU cores and includes multiple channels.
+Although it is necessary to feed channels within their respective time windows separately, you cannot disable an individual channel without disabling the entire Watchdog peripheral.
+Instead, you can disable the entire Watchdog peripheral, and if you wish to re-enable it, you must also restore all other Watchdog sources.
+
+To enable the Global Watchdog module, use the ``Nrf::Watchdog::Enable()`` function:
+
+.. code-block:: c++
+
+    Nrf::Watchdog::Enable();
+
+From this point, all previously created Matter Watchdog objects must be fed periodically to comply with the time window requirement.
+
+To disable the Global Watchdog module, use the ``Nrf::Watchdog::Disable()`` function:
+
+.. code-block:: c++
+
+    Nrf::Watchdog::Disable();
+
+This method disables all previously enabled Matter Watchdog objects, removes their channels, and stops the automatic feeding (if it was configured and enabled).
+
+To enable Matter Watchdog sources again, you need to call the ``Restore()`` method on each of them.
+
+For example:
+
+    .. code-block:: c++
+
+        myWatchdog1.Restore();
+        myWatchdog2.Restore();
+
+End then enable the Global Watchdog module:
+
+.. code-block:: c++
+
+    Nrf::Watchdog::Enable();
+
+Limited lifetime of the Watchdog source
+***************************************
+
+Watchdog source can be created locally and removed when the objects is deleting.
+
+For example you can use it in the specific function:
+
+.. code-block:: c++
+
+    void DoTask()
+    {
+        Nrf::Watchdog::WatchdogSource watchdogSource;
+
+        {
+            while(condition)
+            {
+                /* Do some time-critical operations and break loop */
+                watchdogSource.Feed();
+            }
+        }
+    }
+
+In the function mentioned above, when the Watchdog source object is created, a new Watchdog is also created and it must be fed periodically.
+After performing some time-critical operations and exiting the loop in the ``DoTask`` function, the Watchdog source object is deleted, and its destructor frees the assigned channel.
+
+.. note::
+
+    The maximum number of simultaneously active Watchdog sources is 8. Therefore, creating a ninth Watchdog source will trigger an assert.
```

Comment:
```suggestion
    The maximum number of simultaneously active watchdog sources is 8.
    Creating a ninth watchdog source will trigger an assert.
```
Should this note also be in the creating section?

Reply:
Yes, it should be. Will add it there.